src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticSuppressor.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// The base type for diagnostic suppressors that can programmatically suppress analyzer and/or compiler non-error diagnostics.
+    /// </summary>
+    public abstract class DiagnosticSuppressor : DiagnosticAnalyzer
+    {
+        // Disallow suppressors from reporting diagnostics or registering analysis actions.
+        public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
+
+        public sealed override void Initialize(AnalysisContext context) { }
```

Comment:
>public sealed override void Initialize(AnalysisContext context) { } [](start = 8, length = 67)

Should this delegate to the base? And why not?

Reply:
[DiagnosticAnalyzer.Initialize](http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/DiagnosticAnalyzer.cs,22) is abstract